project/Build.scala

```diff
@@ -1686,6 +1508,35 @@ object Build {
         }
       },
 
+      sourceGenerators in Compile += Def.task {
+        val stage = scalaJSStage.value
+
+        val linkerConfig = stage match {
+          case FastOptStage => (scalaJSLinkerConfig in fastOptJS).value
+          case FullOptStage => (scalaJSLinkerConfig in fullOptJS).value
```

Comment:
I think you need `(scalaJSLinkerConfig in (Compile, fastOptJS)).value` (and same for `fullOptJS`), because the above would always fallback to `scalaJSLinkerConfig.value`, since we only redefine it `in (SomeConfig, fullOptJS)`.